gossip-protocol-spec.md

```diff
@@ -152,29 +192,124 @@ struct Bloom {
 </details>
 
 ### PullResponse
-These are sent in response to a `PullRequest`.
+These are sent in response to a `PullRequest`. They contain filtered values from the node's `crds` which the origin node is missing. 
 
 | Data | Type | Size | Description |
 |------|:----:|:----:|-------------|
 | `Pubkey` | `[u8; 32]` | 32 | a public key of the origin |
 | `CrdsValuesList` | [`[CrdsValue]`](#data-shared-between-nodes) | 8+ | a list of new values  |
 
+<details>
+  <summary>Solana client Rust implementation</summary>
 
+```rust
+enum Protocol {
+    //...
+    PullResponse(Pubkey, Vec<CrdsValue>),
+    //...
+}
+```
+
+</details>
 
-### PushMessage
-It is sent by nodes who want to share information with others. Node receiving the message checks for:
-- duplication - duplicated messages are dropped, and the node responds with a prune message if the message came from a low staked node
-- new data:
-    - new information is stored in `crds` and replaces old value
-    - stores message in `pushed_once`, which is used for detecting duplicates
-    - retransmits information to its peers
-- expiration - messages older than `PUSH_MSG_TIMEOUT` are dropped
+
+### PruneMessage
+It is sent to peers with a list of origin nodes that should be pruned. No more push messages from pruned origin nodes should be sent by the recipient of this prune message to its sender.
 
 | Data | Type | Size | Description |
 |------|:----:|:----:|-------------|
-| `Pubkey` | `[u8; 32]` | 32 | a public key of the origin |
-| `CrdsValuesList` | [`[CrdsValue]`](#data-shared-between-nodes) | 8+ | a list of values to share  |
+| `Pubkey` | `[u8, 32]` | 32 | a public key of the origin |
+| `PruneData` | [`PruneData`](#prunedata) | 144+ | a structure which contains prune details |
+
+
+#### PruneData
+| Data | Type | Size | Description |
+|------|:----:|:----:|-------------|
+| `pubkey` |`[u8, 32]` | 32 | public key of the origin |
+| `prunes` | `[[u8, 32]]` | 8+ | public keys of nodes that should be pruned |
```

Comment:
`node` is technically correct but could say `origin node` or something to be a little more exact